runtime/plaid/src/data/sqs/mod.rs

```diff
@@ -0,0 +1,194 @@
+use std::num::NonZeroUsize;
+
+use aws_sdk_sqs::Client;
+use crossbeam_channel::Sender;
+use lru::LruCache;
+use plaid_stl::messages::{Generator, LogSource, LogbacksAllowed};
+use serde::Deserialize;
+use time::OffsetDateTime;
+
+use crate::{executor::Message, get_aws_sdk_config, AwsAuthentication};
+
+use super::{DataGenerator, DataGeneratorLog};
+
+#[derive(Deserialize)]
+pub struct SQSConfig {
+    /// Name of this SQS queue
+    pub name: String,
+    /// Polling URL for this SQS queue
+    pub queue_url: String,
+    /// Polling interval for this SQS queue
+    pub wait_time_seconds: i32,
+    /// Max number of messsages to get 1 - 10
+    pub max_number_of_messages: i32,
+    /// How to authenticate to AWS
+    pub authentication: AwsAuthentication,
+    /// Denotes if logs produced by this generator are allowed to initiate log backs
+    #[serde(default)]
+    logbacks_allowed: LogbacksAllowed,
+    /// Canonicalization time, i.e., after how many seconds we can consider logs as "stable"
+    #[serde(default = "default_canon_time")]
+    canon_time: u64,
+    /// Number of milliseconds to wait in between calls to the API.
+    /// If no value is provided here, we will use a default value (1 second).
+    #[serde(default = "default_sleep_milliseconds")]
+    sleep_duration: u64,
+}
+
+/// This function provides the default sleep duration in milliseconds.
+/// It is used as the default value for deserialization of the `sleep_duration` field,
+/// of `SQSConfig` in the event that no value is provided.
+fn default_sleep_milliseconds() -> u64 {
+    1000
+}
+
+fn default_canon_time() -> u64 {
+    // guestimate
```

Comment:
Should try and formalize this. But SQS seems like a service that would guarantee ordering?